z390test/src/test/groovy/org/z390/test/TestZ390Test.groovy

```diff
@@ -0,0 +1,50 @@
+package org.z390.test
+
+import org.junit.jupiter.api.Test
+
+class TestZ390Test extends z390Test {
+
+    var sysmac = basePath("mac")
+    var options = ['trace', 'noloadhigh', "SYSMAC(${sysmac})"]
+
+    @Test
+    void testAsm() {
+        int rc = this.asm(basePath("tests", "TESTINS1"), *options)
+        this.printOutput()
+        assert rc == 0
+    }
+
+    @Test
+    void testAsml() {
+        int rc = this.asml(basePath("tests", "TESTINS2"), *options)
+        this.printOutput()
+        assert rc == 0
+    }
+
+    @Test
+    void testAsmlg() {
+        int rc = this.asmlg(basePath("tests", "TESTINS2"), *options)
+        this.printOutput()
+        assert rc == 0
+    }
+
+    @Test
+    void testInlineSource() {
+
+        var source = """TESTB    START 0
+         USING *,13
+         STM   14,12,12(13)
+         ST    13,8(13)
+         ST    15,4(15)
+         LR    13,15
+         J
+         RETURN (14,12)
+         END   TESTB
+    """
```

Comment:
While this sample code will produce the expected error message, it contains several logic errors unrelated the test. Replace it with
```
TESTB    START 0
         STM   14,12,12(13)
         BAS   15,8+72(15)
         USING *,13
         DC    18F'0'
         ST    13,4(15)
         ST    15,8(13)
         LR    13,15
         J
         L     13,4(13)
         RETURN (14,12)        Regs restored  from 12 past "STM ..."
 
         END   TESTB
```
If you put an asterisk in column 1 of the "J" line, running asmlg on the file will run to completion without error. The existing source will not.

Reply:
Have updated the assembler code as suggested. Generates the required error message with the J line included.